File: Laba9/src/Tuner.py
from src.TV import TV
from enum import Enum

class Tuner(TV):

    class Sources(Enum):
        TELEVISION = 0
        ANDROIDTV = 1

    #Constructor
    def __init__(self, ssid = "test", password = "test", type = 0, volume = 10, maxVolume = 100, maxChannelNumber = 100):
        super().__init__(ssid, password, type, volume, maxVolume)
        self.__source = self.Sources.TELEVISION
        self.__channelNumber = 0
        self.__maxChannelNumber = maxChannelNumber

    #Method open app by id
    def openApp(self, appIndex):
        if self.__source == self.Sources.ANDROIDTV:
            print(self._TV__androidTV.openApp(appIndex))
        else:
            print("Change source to Android TV")
        print()

    #Method provides televison data
    def getTVData(self):
        print("Screen mode: " + str(self._TV__screen.getMode()))
        print("Screen type: " + str(self._TV__screen.getType()))
        print("Volume: " + str(self._TV__speakers.getVolume()))
        print("Enabled: " + str(self._TV__isEnabled))
        print("Wifi: " + str(self._TV__androidTV.getWifiStatus()))
        print("Source: " + str(self.__source.name))
        print("Channel Number: " + str(self.__channelNumber))
        print("Opened App: " + str(self._TV__androidTV.getOpenedApp()))
        print()

    #Method change television source
    def changeSource(self, source):
        if (source < 0):
            source = 0
        elif (source > 1):
            source = 1

        self.__source = self.Sources(source)

        print("Source changed to " + self.__source.name)
        print()

    #Method go to next channel
    def nextChannel(self):
        self.changeChannel(self.__channelNumber + 1)

    #Method go to previous channel
    def previousChannel(self):
        self.changeChannel(self.__channelNumber - 1)

    #Method change channel
    def changeChannel(self, channel):
        if self.__source == self.Sources.TELEVISION:
            self.__channelNumber = channel;

            if self.__channelNumber < 0:
                self.__channelNumber = self.__maxChannelNumber;
            elif self.__channelNumber > self.__maxChannelNumber:
                self.__channelNumber = 0

            print("Channel changed to " + str(self.__channelNumber));
        else:
            print("Change source to Televison");
        print()

File: Laba3/src/Tuner.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintWriter;

/**
 * Class <code>Tuner</code> implements Source and extends TV
 */
public class Tuner extends TV implements Source {

    enum Sources {
        TELEVISION, ANDROIDTV
    };

    private Sources source;
    private int channelNumber;
    private int maxChannelNumber;

    /**
     * Constructor
     * 
     * @throws FileNotFoundException
     */
    public Tuner() throws FileNotFoundException {
        androidTV = new AndroidTV();
        screen = new Screen();
        speakers = new Speakers();
        isEnabled = false;
        source = Sources.TELEVISION;
        channelNumber = 0;
        maxChannelNumber = 100;

        file = new PrintWriter(new File("Log.txt"));
    }

    /**
     * Constructor
     * 
     * @param ssid             WIFI SSID
     * @param password         WIFI password
     * @param screenType       Type of screen
     * @param volume           Base television volume
     * @param maxVolume        Max television volume
     * @param maxChannelNumber Max channel number
     * @throws FileNotFoundException
     */
    public Tuner(String ssid, String password, int screenType, int volume, int maxVolume, int maxChannelNumber)
            throws FileNotFoundException {
        androidTV = new AndroidTV(ssid, password);
        screen = new Screen(screenType);
        speakers = new Speakers(volume, maxVolume);
        isEnabled = false;
        source = Sources.TELEVISION;
        channelNumber = 0;
        this.maxChannelNumber = maxChannelNumber;

        file = new PrintWriter(new File("Log.txt"));
    }

    /**
     * Method change televison state from off to on
     */
    public void turnOn() {
        androidTV.connectToWifi();
        isEnabled = true;

        file.println("TV enabled");
        file.println(androidTV.getWifiStatus());
        file.println(androidTV.openApp(0));
        file.println();
        file.flush();
    }

    /**
     * Method change televison state from on to off
     */
    public void turnOff() {
        isEnabled = false;

        file.println("TV disabled");
        file.println();
        file.flush();
    }

    /**
     * Method open app by id
     * 
     * @param app
     */
    public void openApp(int app) {
        if (source == Sources.ANDROIDTV)
            file.println(androidTV.openApp(app));
        else
            file.println("Change source to Android TV");

        file.println();
        file.flush();
    }

    /**
     * Method provides televison data in logs
     */
    public void getTVData() {
        file.println("Screen mode: " + screen.getMode());
        file.println("Screen type: " + screen.getType());
        file.println("Volume: " + speakers.getVolume());
        file.println("Enabled: " + isEnabled);
        file.println("Wifi: " + androidTV.getWifiStatus());
        file.println("Source: " + source);
        file.println("Channel number: " + channelNumber);
        file.println("Opened App: " + androidTV.getOpenedApp());
        file.println();
        file.flush();
    }

    /**
     * Method change television source
     * 
     * @param source
     */
    public void changeSource(int source) {
        if (source < 0) {
            source = 0;
        } else if (source > 1) {
            source = 1;
        }
        this.source = Sources.values()[source];

        file.println("Source changed to " + this.source);

        file.println();
        file.flush();
    }

    /**
     * Method go to next channel
     */
    public void nextChannel() {
        changeChannel(channelNumber + 1);
    }

    /**
     * Method go to previous channel
     */
    public void previousChannel() {
        changeChannel(channelNumber - 1);
    }

    /**
     * Method change channel
     * 
     * @param channel
     */
    public void changeChannel(int channel) {
        if (source == Sources.TELEVISION) {
            channelNumber = channel;

            if (channelNumber < 0) {
                channelNumber = maxChannelNumber;
            } else if (channelNumber > maxChannelNumber) {
                channelNumber = 0;
            }

            file.println("Channel changed to " + channelNumber);
        } else
            file.println("Change source to Televison");

        file.println();
        file.flush();
    }
}
